services/movies.go

```diff
@@ -0,0 +1,67 @@
+package services
+
+import (
+	"database/sql"
+
+	"github.com/RealImage/moviebuff-sdk-go"
+	"github.com/sirupsen/logrus"
+
+	"github.com/ashwinspg/explore-golang/config"
+	"github.com/ashwinspg/explore-golang/daos"
+	"github.com/ashwinspg/explore-golang/dtos"
+	"github.com/ashwinspg/explore-golang/utils"
+)
+
+//Movie - service
+type Movie struct {
+	dao *daos.Movie
+	l   *logrus.Entry
+}
+
+//NewMovie - instance creation
+func NewMovie(db *sql.DB, l *logrus.Entry) *Movie {
+	return &Movie{
+		dao: daos.NewMovie(db),
+		l:   l,
+	}
+}
+
+//GetMovie - get movie based on uuid
+func (m *Movie) GetMovie(uuid string) (movieDTO dtos.Movie, err error) {
+	movieDTO, err = m.dao.FindByID(uuid)
+	switch err {
```

Comment:
Prefer early returns. if err == nil, return https://golang.org/doc/effective_go.html#if

Reply:
please elaborte on this one...because in my case, nil is the first case which will be return early....
also, I make use of reference from slate-core (file: advertisers.go):
```
func GetAdvertiserTypeByID(w http.ResponseWriter, r *http.Request) {
	r, rd := logAndGetRequestData(w, r)
	id := chi.URLParam(r, "id")
	dto, err := services.NewAdvertiserType(rd.db, rd.logger).GetByID(id)
	switch err {
	case nil:
		writeJSONStruct(dto, http.StatusOK, rd)
	case services.ErrAdvertiserTypeNotFound:
		writeJSONMessage(err.Error(), http.StatusNotFound, rd)
	case services.ErrInvalidAdvertiserTypeID:
		writeJSONMessage(err.Error(), http.StatusBadRequest, rd)
	default:
		writeJSONMessage("Failed to get advertiser type"+id, http.StatusInternalServerError, rd)
	}
}
```

Reply:
The idea is not to indent too much, and preserve readability.
In this particular case, err == nil can return straight away. For the specific case of ErrMovieNotFound, we have some logic that is in the middle of a case block whereas this can the last control flow.
```
if err == nil {
   return
}

if err != daos.ErrMovieNotFound {
   m.l.WithError(err).Errorln("Failed to get Movie Information from Database")
   return
}

//Now simply put in the logic for ErrMovieNotFound
```
This is easier to read because we don't have to scroll